cmd/srcd/cmd/prune_test.go

```diff
@@ -0,0 +1,99 @@
+// +build integration
+
+package cmd
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+
+	cmdtest "github.com/src-d/engine/cmd/test-utils"
+	"github.com/src-d/engine/docker"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/suite"
+)
+
+// This test suite does not test the `--with-images` flag.
+// If the tests are run with `make test-integration` the daemon image will be
+// the one build locally (e.g. srcd/cli-daemon:dev-b72f1fe), and deleting
+// this image would make all the other tests fail.
+
+type PruneTestSuite struct {
+	cmdtest.IntegrationSuite
+	testDir string
+}
+
+func TestPruneTestSuite(t *testing.T) {
+	s := PruneTestSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *PruneTestSuite) SetupTest() {
+	var err error
+	s.testDir, err = ioutil.TempDir("", "prune-test")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (s *PruneTestSuite) TearDownTest() {
+	s.RunStop(context.Background())
+	os.RemoveAll(s.testDir)
+}
+
+func (s *PruneTestSuite) TestRunningContainers() {
+	require := s.Require()
+
+	_, err := s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	_, err = s.RunSQL(context.TODO(), "SELECT 1")
+	require.NoError(err)
+
+	_, err = s.RunCommand(context.TODO(), "prune")
+	require.NoError(err)
+
+	// Test containers were deleted
+	s.AllStopped()
+
+	// Test volumes were deleted.
+	// This does not actually test much because Engine does not create volumes
+	// since v0.7.0.
+	// The logic used in prune to delete volumes is looking for the
+	// srcd-cli- prefix in the name, so that's what we check here.
+	vols, err := docker.ListVolumes(context.Background())
+	require.NoError(err)
+
+	for _, vol := range vols {
+		require.NotContainsf(vol.Name, "srcd-cli-", "Volume %s was not deleted", vol.Name)
+	}
+
+	// Test networks were deleted.
+	nets, err := listNetworks()
+	require.NoError(err)
+
+	for _, net := range nets {
+		require.NotEqualf(net.Name, docker.NetworkName, "Network %s was not deleted", net.Name)
```

Comment:
would not be better to ensure that there is no `srcd-cli-network` left after pruning?

Reply:
same as the other comment, I don't understand what you mean. That is what the code is doing, checking `srcd-cli-network` is not in the list of networks after pruning.

Reply:
same shame :1st_place_medal: